src/target/cortexm.c

```diff
@@ -118,103 +117,101 @@ struct cortexm_priv {
-	"    <reg name=\"psp\" bitsize=\"32\" save-restore=\"no\" type=\"data_ptr\"/>"
-	"    <reg name=\"primask\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"basepri\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"faultmask\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"control\" bitsize=\"8\" save-restore=\"no\"/>"
-	"  </feature>"
-	"</target>";
-
-static const char tdesc_cortex_mf[] =
-	"<?xml version=\"1.0\"?>"
-	"<!DOCTYPE target SYSTEM \"gdb-target.dtd\">"
-	"<target>"
-	"  <architecture>arm</architecture>"
-	"  <feature name=\"org.gnu.gdb.arm.m-profile\">"
-	"    <reg name=\"r0\" bitsize=\"32\"/>"
-	"    <reg name=\"r1\" bitsize=\"32\"/>"
-	"    <reg name=\"r2\" bitsize=\"32\"/>"
-	"    <reg name=\"r3\" bitsize=\"32\"/>"
-	"    <reg name=\"r4\" bitsize=\"32\"/>"
-	"    <reg name=\"r5\" bitsize=\"32\"/>"
-	"    <reg name=\"r6\" bitsize=\"32\"/>"
-	"    <reg name=\"r7\" bitsize=\"32\"/>"
-	"    <reg name=\"r8\" bitsize=\"32\"/>"
-	"    <reg name=\"r9\" bitsize=\"32\"/>"
-	"    <reg name=\"r10\" bitsize=\"32\"/>"
-	"    <reg name=\"r11\" bitsize=\"32\"/>"
-	"    <reg name=\"r12\" bitsize=\"32\"/>"
-	"    <reg name=\"sp\" bitsize=\"32\" type=\"data_ptr\"/>"
-	"    <reg name=\"lr\" bitsize=\"32\" type=\"code_ptr\"/>"
-	"    <reg name=\"pc\" bitsize=\"32\" type=\"code_ptr\"/>"
-	"    <reg name=\"xpsr\" bitsize=\"32\"/>"
-	"    <reg name=\"msp\" bitsize=\"32\" save-restore=\"no\" type=\"data_ptr\"/>"
-	"    <reg name=\"psp\" bitsize=\"32\" save-restore=\"no\" type=\"data_ptr\"/>"
-	"    <reg name=\"primask\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"basepri\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"faultmask\" bitsize=\"8\" save-restore=\"no\"/>"
-	"    <reg name=\"control\" bitsize=\"8\" save-restore=\"no\"/>"
-	"  </feature>"
-	"  <feature name=\"org.gnu.gdb.arm.vfp\">"
-	"    <reg name=\"fpscr\" bitsize=\"32\"/>"
-	"    <reg name=\"d0\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d1\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d2\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d3\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d4\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d5\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d6\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d7\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d8\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d9\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d10\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d11\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d12\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d13\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d14\" bitsize=\"64\" type=\"float\"/>"
-	"    <reg name=\"d15\" bitsize=\"64\" type=\"float\"/>"
-	"  </feature>"
-	"</target>";
+static const char tdesc_cortex_m[] = "<?xml version=\"1.0\"?>"
```

Comment:
Suggest putting this in a `/* clang-format: off */` block and formatting it so the indentation is in the source but not inside the strings. This will save us some space int the firmware and make it easier to read

Reply:
fyi `/* clang-format: off */` doesnt work, must be `/* clang-format off */` TIL too